File: php/src/ast/statements/ReturnStatement.php
<?php

require 'vendor/autoload.php';

readonly class ReturnStatement implements Statement
{
    public function __construct(
        public Token $token,
        public Expression $returnValue
    ) {
    }

    public function statementNode(): void
    {
    }

    public function tokenLiteral(): string
    {
        return $this->token->literal;
    }

    public function __toString(): string
    {
        return "return " . $this->returnValue . ";";
    }
}

File: java/monkey/src/main/java/root/ast/statements/ReturnStatement.java
package root.ast.statements;

import root.LocalizedToken;
import root.ast.expressions.Expression;

public class ReturnStatement extends Statement {

    private Expression returnValue;

    public ReturnStatement(LocalizedToken token) {
        super(token);
    }

    public Expression getReturnValue() {
        return returnValue;
    }

    public void setReturnValue(Expression returnValue) {
        this.returnValue = returnValue;
    }

    @Override
    public String stringRep() {
        return "%s %s;".formatted(tokenLiteral(), returnValue.stringRep());
    }
}
